### DISCUSSION TITLE
Reorder List Items

### DISCUSSION THREAD

**Discussion by linaspurinis** [trakt/trakt-api#257] [Q&A] [answered]
I have a question regarding API call Reorder List Items https://api.trakt.tv/users/id/lists/reorder

What is the fastest way to sort the large list after adding a single item to it?
From the documentation I understand, what in order to sort I have two options:

Reread the whole list (since the newly added item would have some unique ID assigned to it which I would need for sorting, so I have to reread the whole list and match the ids). Match, sort post the reorder API.
Delete the whole list and read all items in the correct order.
Both seem like unnecessary heavy work for a single task. Why I cannot sort using traktid? I would be able to add a single item and post the reorder IDs list via API. But API call requires a unique number for the list item which is unknown to me...

Thank you.

**Comment by rectifyer** [comment, marked as answer]
Deleting the list then re-adding it all definitely shouldn't be done, that is too destructive of an action.

I'd go with the first option, and you might already have the list item data cached to use?
